include/boost/numeric/odeint/stepper/detail/adaptive_adams_coefficients.hpp

```diff
@@ -0,0 +1,165 @@
+#ifndef BOOST_NUMERIC_ODEINT_STEPPER_DETAIL_ADAPTIVE_ADAMS_COEFFICIENTS_HPP_INCLUDED
+#define BOOST_NUMERIC_ODEINT_STEPPER_DETAIL_ADAPTIVE_ADAMS_COEFFICIENTS_HPP_INCLUDED
+
+#include <boost/numeric/odeint/stepper/detail/rotating_buffer.hpp>
+
+#include <boost/numeric/odeint/util/state_wrapper.hpp>
+#include <boost/numeric/odeint/util/is_resizeable.hpp>
+#include <boost/numeric/odeint/util/resizer.hpp>
+
+#include <boost/numeric/odeint/util/unwrap_reference.hpp>
+#include <boost/numeric/odeint/util/bind.hpp>
+
+#include <boost/numeric/odeint/algebra/algebra_dispatcher.hpp>
+#include <boost/numeric/odeint/algebra/operations_dispatcher.hpp>
+
+#include <boost/array.hpp>
+
+namespace boost {
+namespace numeric {
+namespace odeint {
+namespace detail {
+
+template<
+size_t Steps,
+class Deriv,
+class Value = double,
+class Time = double,
+class Algebra = typename algebra_dispatcher< Deriv >::algebra_type,
+class Operations = typename operations_dispatcher< Deriv >::operations_type,
+class Resizer = initially_resizer
+>
+class adaptive_adams_coefficients
+{
+public:
+    static const size_t steps = Steps;
+
+    typedef unsigned short order_type;
+    static const order_type order_value = steps;
+
+    typedef Value value_type;
+    typedef Deriv deriv_type;
+    typedef Time time_type;
+
+    typedef state_wrapper< deriv_type > wrapped_deriv_type;
+    typedef rotating_buffer< wrapped_deriv_type , steps+1 > step_storage_type; // +1 for moulton
+    typedef rotating_buffer< time_type , steps+1 > time_storage_type;
+
+    typedef Algebra algebra_type;
+    typedef Operations operations_type;
+    typedef Resizer resizer_type;
+
+    typedef adaptive_adams_coefficients< Steps , Deriv , Value , Time , Algebra , Operations , Resizer > aac_type;
+
+    adaptive_adams_coefficients( const algebra_type &algebra = algebra_type())
+    :m_eo(1), m_steps_init(1), beta(), phi(), m_ns(0), m_time_storage(),
+    m_algebra(algebra),
+    m_phi_resizer()
+    {
+        for (size_t i=0; i<order_value+2; ++i)
+        {
+            c[0][i] = 1.0/(i+1);
+        }
+
+        g[0] = c[0][0];
+
+        beta[0][0] = 1;
+        beta[1][0] = 1;
+    };
+
+    void predict(time_type t, time_type dt)
+    {
+        m_time_storage[0] = t;
+
+        if (fabs(m_time_storage[0] - m_time_storage[1] - dt) > 1e-16 || m_eo >= m_ns)
```

Comment:
I added adaptive_adams_bashforth_moulton and controlled_adams_bashforth_moulton to the tests in integrate.cpp, so it should work.